src/components/plots/PlotContainer.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  Collapse, Button, Skeleton, Space, Tooltip,
+} from 'antd';
+import { InfoCircleOutlined } from '@ant-design/icons';
+
+import { LOAD_CONFIG } from 'redux/actionTypes/componentConfig';
+
+import { savePlotConfig } from 'redux/actions/componentConfig/index';
+import { initialPlotConfigStates } from 'redux/reducers/componentConfig/initialState';
+import _ from 'lodash';
+
+const { Panel } = Collapse;
+
+const ResetButton = ({ onClickReset, disabled }) => (
+  <Button
+    key='reset'
+    type='primary'
+    size='small'
+    onClick={onClickReset}
+    disabled={disabled}
+  >
+    Reset
+  </Button>
+);
+
+ResetButton.propTypes = {
+  onClickReset: PropTypes.func.isRequired,
+  disabled: PropTypes.bool.isRequired,
+};
+
+const PlotContainer = (props) => {
+  const {
+    experimentId,
+    plotUuid, plotType, plotInfo,
+    extra, showReset, children,
+  } = props;
+
+  const dispatch = useDispatch();
+
+  const [enableReset, setEnableReset] = useState(true);
+  const { config } = useSelector((state) => state.componentConfig[plotUuid] || {});
+
+  const checkConfigEquality = (currentConfig, initialConfig) => {
+    const ignoredFields = {
+      // config fields that are set dynamically on component render
+      // should not be compared to their initial values
+      embeddingContinuous: ['shownGene'],
+      violin: ['shownGene', 'title'],
+      markerHeatmap: ['selectedGenes'],
+      DotPlot: ['selectedGenes'],
+    };
```

Comment:
Hmm... I'm not sure how to make this more automatic. As commented, these are fields which are dynamically changed. Most of them are obvious (e.g. `shownGene`, `selectedGenes`), but one example (`title` on violin plot) isn't. Putting them inside a property like you mention is a good start,  but whoever is creating the plot would also have to be aware. Maybe We can have an array of keys in each plot's config that we want to skip when we reset the config  (e.g. `keepValuesOnReset`). That would be a start I think.

Reply:
If we use the object, then we'd have to be aware that for some plots, keys that are usually under config (e.g. `title`) would be under the object key (e.g. if the object is `customFields` then the value would be in `customFields.title`). 

It would also make it difficult if we want to make changes to the field which makes it not dynamic anymore. For example, if in the previous case, if we don't want `title` to not be dynamic anymore, we'd have to move `title` out and make changes to the corresponding `generatePlotConfig/generatePlotData` functions.

If we use the array, we can just insert/remove keys which we want to ignore.

Reply:
Ah ok, good points, array sounds good then